report.md

```diff
@@ -2,6 +2,24 @@
 
 ## Subsystem 1 (MFA)
 
+Multifactor authentication will apply to all users and will aim for a balance between confidentiality and availability.
+
+### First Factor
+The first step of authentication will involve a username and password. The username will be stored as a hash. The password will be stored as a hash with salt. The salt for each user will be unique and will be generated upon registration. The salt will be stored as plain text.
+
+### Second Factor
+The second factor will be triggered on the completion of the first factor. On passing the first stage a random string will be generated by the server and stored temporarily in the database with the associated user in hash form. An sms will be sent to the user based on the registered phone number with the digits. The user will be redirected to another form to key in the digits. He will have 2 minutes to do so after which the login process will fail and the temporary string will be replaced with nonces. On success the user will receive a session and be able to login and the string will be replaced with nonces.

```

Comment:
Checked with Prof on this 2nd FA. Prof is ok with either. 

I'm thinking we can take this out of our report (in case we don't have time to implement it :P), but if we do have, then ok la we can implement it haha.

Reply:
okay. do you want to take it out or leave it but say something like it depends on time constraint